From 5f95fc3f2e4684139cc0acbfe5a2fd6104f97352 Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Sun, 4 Apr 2010 06:24:41 -0500
Subject: [PATCH 07/32] v4l2: clear flags before reusing buffer from buffer pool

note: this really only effects v4l2sink since gst_v4l2_buffer_pool_get() is
only called once per buffer in the v4l2src case (in
gst_v4l2src_buffer_pool_activate())
---
 sys/v4l2/gstv4l2bufferpool.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/sys/v4l2/gstv4l2bufferpool.c b/sys/v4l2/gstv4l2bufferpool.c
index c13bb1e..9bd63a6 100644
--- a/sys/v4l2/gstv4l2bufferpool.c
+++ b/sys/v4l2/gstv4l2bufferpool.c
@@ -461,8 +461,10 @@ gst_v4l2_buffer_pool_get (GstV4l2BufferPool * pool)
 {
   GstV4l2Buffer *buf = g_async_queue_try_pop (pool->avail_buffers);
 
-  if (buf)
+  if (buf) {
     GST_BUFFER_SIZE (buf) = buf->vbuffer.length;
+    GST_BUFFER_FLAG_UNSET (buf, 0xffffffff);
+  }
 
   pool->running = TRUE;
 
-- 
1.7.1

